### ISSUE TITLE
Will there be an update?

### ISSUE THREAD

**Issue by MajorP93** [haiwen/seafile-rpi#118]
Hello folks,

I'd like to know wether or not we can expect an updated seafile package for the Raspberry Pi?

The latest Seafile version is 9.0.8 while this Raspberry Pi package is still at 9.0.2.

Looking forward to hearing from you.

Thanks!

**Comment by Gustl22** [comment]
I wish I could invest more time to debug. I'll see what I can do :D

**Comment by nihilistaX** [comment]
Running on self-build 9.06 here, but after 9.07 build can't get it working. Builds fine, starting without error but no connection possible, maybe some bad dependency....

**Comment by gu3nter** [comment]
Running seafile on self build 9.0.9 since yesterday on Ubuntu 20.4 64bit without any issues so far. I only had to add markupsafe==2.0.1 to geht seafdav working.
If someone wants i can send a Link

**Comment by MajorP93** [comment]
> Running seafile on self build 9.0.9 since yesterday on Ubuntu 20.4 64bit without any issues so far. I only had to add markupsafe==2.0.1 to geht seafdav working. If someone wants i can send a Link

Where did you add "markupsafe==2.0.1"?

Can you elaborate your steps for building 9.0.9?

Thanks!

**Comment by gu3nter** [comment]
I´ve done the following steps:
I created an own version of https://raw.githubusercontent.com/haiwen/seafdav/v9.0.9-server/requirements.txt in my current seafile and added a new line with "markupsafe==2.0.1" at the end of the file to set the link for the build script, see last step

1. wget https://raw.githubusercontent.com/haiwen/seafile-rpi/master/build.sh
2. chmod u+x build.sh
3. python -m venv env && source env/bin/activate
4. pip install --upgrade pip
5. pip install wheel
6. pip install distro

7. ./build.sh -DTA -v 9.0.9 -h https://raw.githubusercontent.com/haiwen/seahub/v9.0.9-server/requirements.txt -d https://my-own-generated-seafdav.txt

thats all

Note: If python 2 is the default python version in your setup you have to execute python3 -m venv env && source env/bin/activate in step 3

**Comment by MajorP93** [comment]
I followed @gu3nter steps but had comment out `cargo` and `golang-go` from the "apt install" build dependencies part in the build.sh script.

Reason is I am running raspbian 10 (buster) and cargo/rust and golang versions in the apt repositories are simply too old for seafile to compile succesfully.

After commenting those 2 packages out I installed correct versions of cargo, rustc and golang manually.
Then I was able to succesfully build seafile 9.0.9, install it and start the services afterwards.

Thanks for pointing me into the right direction @gu3nter .

If somebody else is using raspbian 10 (buster) and wants me to upload my `seafile-server-9.0.9-buster-armv7l.tar.gz` package, I will be happy to do so.

Best regards

**Comment by nihilistaX** [comment]
> I followed @gu3nter steps but had comment out `cargo` and `golang-go` from the "apt install" build dependencies part in the build.sh script.
> 
> Reason is I am running raspbian 10 (buster) and cargo/rust and golang versions in the apt repositories are simply too old for seafile to compile succesfully.
> 
> After commenting those 2 packages out I installed correct versions of cargo, rustc and golang manually. Then I was able to succesfully build seafile 9.0.9, install it and start the services afterwards.
> 
> Thanks for pointing me into the right direction @gu3nter .
> 
> If somebody else is using raspbian 10 (buster) and wants me to upload my `seafile-server-9.0.9-buster-armv7l.tar.gz` package, I will be happy to do so.
> 
> Best regards

Hey, would be fine, just for testing as my own build did start, but webinterface is not accessible....

**Comment by MajorP93** [comment]
> > I followed @gu3nter steps but had comment out `cargo` and `golang-go` from the "apt install" build dependencies part in the build.sh script.
> > Reason is I am running raspbian 10 (buster) and cargo/rust and golang versions in the apt repositories are simply too old for seafile to compile succesfully.
> > After commenting those 2 packages out I installed correct versions of cargo, rustc and golang manually. Then I was able to succesfully build seafile 9.0.9, install it and start the services afterwards.
> > Thanks for pointing me into the right direction @gu3nter .
> > If somebody else is using raspbian 10 (buster) and wants me to upload my `seafile-server-9.0.9-buster-armv7l.tar.gz` package, I will be happy to do so.
> > Best regards
> 
> Hey, would be fine, just for testing as my own build did start, but webinterface is not accessible....

Hey!
Sure, you can find my compiled package here:
https://github.com/MajorP93/seafile-rpi/releases/download/v9.0.9/seafile-server-9.0.9-buster-armv7l.tar.gz

By the way: for me the web interface is working without any issues.

**Comment by shadowempire123** [comment]
any word about an officially updated raspi version?

uname -a
Linux seafile 5.10.103-v7+ #1529 SMP Tue Mar 8 12:21:37 GMT 2022 armv7l GNU/Linux

lsb_release -a
No LSB modules are available.
Distributor ID: Raspbian
Description:    Raspbian GNU/Linux 10 (buster)
Release:        10
Codename:       buster

**Comment by ChatDeBlofeld** [comment]
Hi @shadowempire123,

This version is community maintained, if there is no one to build the packages, well...

**Comment by shadowempire123** [comment]
I tried to update seafile server 9.0.2 to 9.0.9

```
sudo su
cd /home/seafile
wget https://github.com/MajorP93/seafile-rpi/releases/download/v9.0.9/seafile-server-9.0.9-buster-armv7l.tar.gz
tar -xzf seafile-server-9.0.9-buster-armv7l.tar.gz
systemctl stop seafile seahub
cd seafile-server-9.0.9/upgrade
./upgrade_8.0_9.0.sh
systemctl start seafile seahub

```

with the following result:



Page unavailable
Sorry, but the requested page is unavailable due to a server hiccup.

Our engineers have been notified, so check back later.

seafile.log:
2022-11-16 09:41:58 failed to bind unix socket fd to /home/seafile/seafile-server-9.0.9/runtime/seafile.sock : Permission denied
2022-11-16 09:41:58 seaf-server.c(1070): Failed to start rpc server.

controller.log:
2022-11-16 09:49:32 seafile-controller.c(447): pid file /home/seafile/pids/seaf-server.pid does not exist
2022-11-16 09:49:32 seafile-controller.c(513): seaf-server need restart...
2022-11-16 09:49:32 seafile-controller.c(191): starting seaf-server ...
2022-11-16 09:49:32 seafile-controller.c(82): spawn_process: seaf-server -F /home/seafile/conf -c /home/seafile/ccnet -d /home/seafile/seafile-data -l /home/seafile/logs/seafile.log -P /home/seafile/pids/seaf-server.pid -p /home/seafile/seafile-server-9.0.9/runtime
2022-11-16 09:49:32 seafile-controller.c(116): spawned seaf-server, pid 1369
2022-11-16 09:49:42 seafile-controller.c(447): pid file /home/seafile/pids/seaf-server.pid does not exist
2022-11-16 09:49:42 seafile-controller.c(513): seaf-server need restart...
2022-11-16 09:49:42 seafile-controller.c(191): starting seaf-server ...
2022-11-16 09:49:42 seafile-controller.c(82): spawn_process: seaf-server -F /home/seafile/conf -c /home/seafile/ccnet -d /home/seafile/seafile-data -l /home/seafile/logs/seafile.log -P /home/seafile/pids/seaf-server.pid -p /home/seafile/seafile-server-9.0.9/runtime
2022-11-16 09:49:42 seafile-controller.c(116): spawned seaf-server, pid 1371


seahub.log:
Traceback (most recent call last):
  File "/home/seafile/seafile-server-9.0.9/seahub/thirdpart/django/core/handlers/exception.py", line 47, in inner
    response = get_response(request)
  File "/home/seafile/seafile-server-9.0.9/seahub/thirdpart/django/core/handlers/base.py", line 181, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/home/seafile/seafile-server-9.0.9/seahub/thirdpart/django/views/decorators/csrf.py", line 54, in wrapped_view
    return view_func(*args, **kwargs)
  File "/home/seafile/seafile-server-9.0.9/seahub/thirdpart/django/views/generic/base.py", line 70, in view
    return self.dispatch(request, *args, **kwargs)
  File "/home/seafile/seafile-server-9.0.9/seahub/seahub/api2/base.py", line 23, in dispatch
    response = super(APIView, self).dispatch(*a, **kw)
  File "/home/seafile/seafile-server-9.0.9/seahub/thirdpart/rest_framework/views.py", line 505, in dispatch
    response = self.handle_exception(exc)
  File "/home/seafile/seafile-server-9.0.9/seahub/seahub/api2/base.py", line 20, in handle_exception
    return super(APIView, self).handle_exception(exc)
  File "/home/seafile/seafile-server-9.0.9/seahub/thirdpart/rest_framework/views.py", line 465, in handle_exception
    self.raise_uncaught_exception(exc)
  File "/home/seafile/seafile-server-9.0.9/seahub/thirdpart/rest_framework/views.py", line 476, in raise_uncaught_exception
    raise exc
  File "/home/seafile/seafile-server-9.0.9/seahub/thirdpart/rest_framework/views.py", line 493, in dispatch
    self.initial(request, *args, **kwargs)
  File "/home/seafile/seafile-server-9.0.9/seahub/thirdpart/rest_framework/views.py", line 410, in initial
    self.perform_authentication(request)
  File "/home/seafile/seafile-server-9.0.9/seahub/thirdpart/rest_framework/views.py", line 324, in perform_authentication
    request.user
  File "/home/seafile/seafile-server-9.0.9/seahub/thirdpart/rest_framework/request.py", line 220, in user
    self._authenticate()
  File "/home/seafile/seafile-server-9.0.9/seahub/thirdpart/rest_framework/request.py", line 373, in _authenticate
    user_auth_tuple = authenticator.authenticate(self)
  File "/home/seafile/seafile-server-9.0.9/seahub/seahub/api2/authentication.py", line 66, in authenticate
    ret = self.authenticate_v2(request, key)
  File "/home/seafile/seafile-server-9.0.9/seahub/seahub/api2/authentication.py", line 102, in authenticate_v2
    user = User.objects.get(email=token.user)
  File "/home/seafile/seafile-server-9.0.9/seahub/seahub/base/accounts.py", line 95, in get
    emailuser = ccnet_threaded_rpc.get_emailuser(email)
  File "/home/seafile/seafile-server-9.0.9/seafile/lib/python3/site-packages/pysearpc/client.py", line 125, in newfunc
    ret_str = self.call_remote_func_sync(fcall_str)
  File "/home/seafile/seafile-server-9.0.9/seafile/lib/python3/site-packages/pysearpc/named_pipe.py", line 97, in call_remote_func_sync
    transport = self._get_transport()
  File "/home/seafile/seafile-server-9.0.9/seafile/lib/python3/site-packages/pysearpc/named_pipe.py", line 87, in _get_transport
    transport = self._create_transport()
  File "/home/seafile/seafile-server-9.0.9/seafile/lib/python3/site-packages/pysearpc/named_pipe.py", line 80, in _create_transport
    transport.connect()
  File "/home/seafile/seafile-server-9.0.9/seafile/lib/python3/site-packages/pysearpc/named_pipe.py", line 44, in connect
    self.pipe.connect(self.socket_path)
FileNotFoundError: [Errno 2] No such file or directory


any idea how to get it work?
thx!

**Comment by nihilistaX** [comment]
Which system? Sure you have the right package for your system?

**Comment by shadowempire123** [comment]
> 

uname -a
Linux seafile 5.10.103-v7+ #1529 SMP Tue Mar 8 12:21:37 GMT 2022 armv7l GNU/Linux

lsb_release -a
No LSB modules are available.
Distributor ID: Raspbian
Description: Raspbian GNU/Linux 10 (buster)
Release: 10
Codename: buster

**Comment by nihilistaX** [comment]
OK, so maybe some missed dependencies, but didn't know much about python....

**Comment by ChatDeBlofeld** [comment]
```
seafile.log:
2022-11-16 09:41:58 failed to bind unix socket fd to /home/seafile/seafile-server-9.0.9/runtime/seafile.sock : Permission denied
2022-11-16 09:41:58 seaf-server.c(1070): Failed to start rpc server.
```

Don't know what's wrong with permissions but there is something to dig here. Does the user running seafile owns this directory?

**Comment by shadowempire123** [comment]
just went back to 9.0.2 via (it is now working again):

```
systemctl stop seafile seahub
cd seafile-server-9.0.2/upgrade/
./upgrade_8.0_9.0.sh
systemctl start seafile seahub

```
Regarding permissions ->
![image](https://user-images.githubusercontent.com/26248866/202170042-d3dbf069-2ec8-4db0-9e7a-1a3955a1809d.png)

**Comment by gu3nter** [comment]
from my point of view you are doing the update not proccess not correct, please have a lokk in seafile manual for corret updateing

Some hint you have to do a minor-upgrade if you are switching from 9.0.2 to 9.0.9
therefore upgrade/minor-upgrade.sh shalle be used instead of 
upgrade/upgrade_8.0_9.0.sh

**Comment by shadowempire123** [comment]
I tried the upgrade now with minor-upgrade with the same result...

**Comment by gu3nter** [comment]
I had this issues some upgrades before...

"Sorry, but the requested page is unavailable due to a server hiccup."
It is due to the fact, that you have executed the update as su, not as the user with which you have installed it in the past, maybe seafile

delete the folder 9.0.9 and redo every step with the correct user

**Comment by ChatDeBlofeld** [comment]
`seafile-server-9.0.9` is owned by `root`, chown it to `seafile` and it should work.

**Comment by shadowempire123** [comment]
👍 thx that worked

**Comment by nihilistaX** [comment]
> Hey! Sure, you can find my compiled package here: https://github.com/MajorP93/seafile-rpi/releases/download/v9.0.9/seafile-server-9.0.9-buster-armv7l.tar.gz
> 
> By the way: for me the web interface is working without any issues.

Thanks for that, now its working.
But on my other device (rpi/64bit OS) its the same: Build running fine, but web interface not accessible with "internal server error". Nothing in logs, so hard to see whats missing. Maybe i missed some seahub requirement.....could you post your requirements.text? Maybe that helps....

Edit: Got it...there are 2 requirements (Markup and bleach) missing in my requirements.txt, so i rebuild and now its working. Strange, seems that are new requirements, until 9.02 i never needed. Btw, seems to work fine now.

**Comment by Gustl22** [comment]
I built for bullseye, focal and jammy:
https://cloud.reb0.org/d/3a3af571a79e4d62a11b/

Still having problems with buster and kinetic. I'll make a release as soon as I make these working.
Thanks again @ChatDeBlofeld

**Comment by Gustl22** [comment]
I [released 9.0.9](https://github.com/haiwen/seafile-rpi/releases/tag/v9.0.9), unfortunately I didn't manage to get buster working. If anyone has the time and effort to provide a working bundle, let me know. Have fun.
